test-apps/ui-test-app/src/frontend/appui/imodelopen/IModelOpen.tsx

```diff
@@ -86,26 +98,26 @@ export class IModelOpen extends React.Component<IModelOpenProps, IModelOpenState
   public async getIModels(iTwinId: string, top: number, skip: number): Promise<IModelInfo[]> {
 
     const accessToken = await IModelApp.getAccessToken();
-    const hubAccess = new IModelHubFrontend();
+    const authorization = async () => toAuthorization(accessToken);
 
     const iModelInfos: IModelInfo[] = [];
-    const queryOptions = new IModelQuery();
-    queryOptions.select("*").top(top).skip(skip);
+    if (!SampleAppIModelApp.hubClient)
+      return iModelInfos;
+
     try {
-      const iModels: HubIModel[] = await hubAccess.hubClient.iModels.get(accessToken, iTwinId, queryOptions);
-      for (const imodel of iModels) {
-        const versions: Version[] = await hubAccess.hubClient.versions.get(accessToken, imodel.id!, new VersionQuery().select("Name,ChangeSetId").top(1));
-        if (versions.length > 0) {
-          imodel.latestVersionName = versions[0].name;
-          imodel.latestVersionChangeSetId = versions[0].changeSetId;
-        }
-      }
-      for (const thisIModel of iModels) {
+      for await (const imodel of SampleAppIModelApp.hubClient?.iModels.getRepresentationList({
+        urlParams: {
+          projectId: iTwinId,
+          $skip: skip,
```

Comment:
This will not work - the async iterator returns all the items in collection, the $skip and $top params allow to finetune the underlying logic about the page size and initial skip. The lack of documentation for the clients does not help here either :) I have thought about exposing raw pages in async iterator and this case proves that it is necessary. For now something like
```
const iModelsIterator = SampleAppIModelApp.hubClient?.iModels.getRepresentationList({
  urlParams: { projectId: iTwinId, $skip: skip, $top: top },
  authorization,
}));
const iModelsPage = await take(iModelsIterator, top);
const iModelInfos = page.map(...)
```
should work.

Reply:
Update: the pages are now exposed via the returned iterator:
https://github.com/iTwin/imodels-clients/blob/5966c1907e2de6001fc7762353d517553e9bf8cd/clients/imodels-client-management/src/operations/imodel/IModelOperations.ts#L33
https://github.com/iTwin/imodels-clients/blob/main/clients/imodels-client-management/src/base/iterators/EntityListIterator.ts
Sample usage:
https://github.com/iTwin/imodels-clients/blob/5966c1907e2de6001fc7762353d517553e9bf8cd/tests/imodels-clients-tests/src/management/iterators/EntityListIteratorImpl.test.ts#L31

Reply:
I just simplified the logic here. It was only used to get the first 80 iModels and wasn't even using the skip functionality so for simplicity just removed it. Now the page size will be set to 80 and we'll only need to make a single request.